src/Composer.vala

```diff
@@ -682,13 +678,45 @@ public class Mail.Composer : Hdy.ApplicationWindow {
 
                     message_content += "<br/><br/>";
                     message_content += content_to_quote;
+
+                    if (message.has_attachment ()) {
+                        get_attachments.begin (message.content);
+                    }
                 }
             }
 
             web_view.set_body_content (message_content);
         }
     }
 
+    private async void get_attachments (Camel.DataWrapper message_content) {
+        if (message_content is Camel.Multipart) {
+            unowned var content = (Camel.Multipart)message_content;
+            for (uint i = 0; i < content.get_number (); i++) {
+                unowned var part = content.get_part (i);
+                unowned var field = part.get_mime_type_field ();
+                if (part.disposition == "attachment") {
+                    try {
+                        var file = File.new_for_path (Path.build_filename (Environment.get_tmp_dir (), part.get_filename ()));
```

Comment:
I would create a temporary directory for each forwarding:

before the for loop:
```
GLib.File? tmp_dir = null;
```

here:
```suggestion
                        if (tmp_dir == null) {
                            tmp_dir = yield GLib.File.new_tmp_dir_async (null, GLib.Priority.DEFAULT, null);
                        }

                        var file = tmp_dir.get_child (part.get_filename ());
```
That way it is safe to forward several mails with the same filename.

Reply:
`new_tmp_dir_async` is only available in GLib >= 2.74
So I used `GLib.File.new_for_path (Path.build_filename (Environment.get_tmp_dir (), Uuid.string_random ()))` with `make_directory` or is there an easier way?

Reply:
Can't you increase the version requirement since this is provided via flatpak?

Reply:
Unfortunately Mail isn't provided as flatpak because evolution data server doesn't 100% work when flatpaked

Reply:
Sorry for the delay but you should use:
```suggestion
                            tmp_dir = GLib.File.new_for_path (GLib.DirUtils.make_tmp (null));
```
instead of `Uuid.string_random ()` which takes care of verifying that the directory doesn't exist previously.